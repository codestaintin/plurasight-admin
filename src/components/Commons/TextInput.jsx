import React from 'react';
import PropTypes from 'prop-types';

const TextInput = ({
  name, label, onChange, placeholder, value, error, pattern
}) => {
  let wrapperClass = 'form-group';
  if (error && error.length > 0) {
    /*eslint-disable */
    wrapperClass += ' ' + 'has-error';
  }

  return (
    <div className={wrapperClass}>
      <label htmlFor={name}>{label}</label>
      <div className="field">
        <input
          type="text"
          name={name}
          className="form-control"
          placeholder={placeholder}
          pattern={pattern}
          value={value}
          onChange={onChange} />
        {error && <div className="alert alert-danger">{error}</div>}
      </div>
    </div>
  );
};

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  pattern: PropTypes.string,
  value: PropTypes.string,
  error: PropTypes.string
};

export default TextInput;
