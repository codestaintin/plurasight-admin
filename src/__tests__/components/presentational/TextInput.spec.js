import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import TextInput from '../../../components/Commons/TextInput';

configure({ adapter: new Adapter() });

describe('<TextInput>', () => {
  const props = {
    name: '',
    label: '',
    onChange: jest.fn(() => {})
  };

  it('renders the <TextInput> component', () => {
    const tree = shallow(<TextInput {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
