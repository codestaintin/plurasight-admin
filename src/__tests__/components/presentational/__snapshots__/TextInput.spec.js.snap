// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput> renders the <TextInput> component 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor=""
  />
  <div
    className="field"
  >
    <input
      className="form-control"
      name=""
      onChange={[MockFunction]}
      type="text"
    />
  </div>
</div>
`;
